src/main/scala/esmeta/phase/RangeTypeCheck.scala

```diff
@@ -0,0 +1,74 @@
+package esmeta.phase
+
+import esmeta.*
+import esmeta.cfg.CFG
+import esmeta.analyzer.*
+import esmeta.analyzer.domain
+import esmeta.util.*
+import esmeta.util.SystemUtils.*
+import esmeta.spec.Spec.*
+import scala.io.StdIn
+import esmeta.error.TypeCheckFail
+import esmeta.error.ESMetaError
+
+case object RangeTypeCheck extends Phase[Unit, Unit] {
+  val name = "range-tycheck"
+
+  val help =
+    "performs a type analysis of ECMA-262 through multiple versions of specification."
+
+  def apply(
+    unit: Unit,
+    cmdConfig: CommandConfig,
+    config: Config,
+  ): Unit =
+    val range =
+      config.range.getOrElse(throw new Exception("range is not specified."))
+    val targets = getAllTargets(range)
+    if (!config.yes)
+      val confirm = StdIn.readLine(
+        s"Are you sure to run analysis for ${targets.length} revisions? (y/n): ",
+      )
+      if (confirm.take(1) != "y")
+        return println("Canceled.")
+
+    for ((target, idx) <- targets.zipWithIndex) {
+      ANALYZE_LOG_DIR = s"$LOG_DIR/ranges/$target"
+      val Version(name, _) = getVersion(target)
+      print(s"[${idx + 1}/${targets.length}]: $name (hash: $target)")
+      try {
+        CmdTypeCheck(
+          List(
+            "-silent",
+            s"-extract:target=$target",
+            "-tycheck:log",
+          ),
+        )
+      } catch {
+        case TypeCheckFail(_, size) => println(s" - $size mismatches occurred")
+        case e: ESMetaError =>
+          Console.err.println("Unexpected error occurred:")
+          throw e
+      }
+    }
+
+    if (!cmdConfig.silent) println("Done.")
+
+  def defaultConfig: Config = Config()
+  val options: List[PhaseOption[Config]] = List(
+    (
+      "range",
+      StrOption((c, s) => c.range = Some(s)),
+      "revision range for ECMA-262.",
+    ),
+    (
+      "yes",
+      StrOption((c, s) => c.yes = true),
```

Comment:
I guess you usually use BoolOption for this kind of use case.

Reply:
I think I missed that. thanks :)